src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,129 @@
+        Node<K, V> newNode = new Node<>(key, value);
+        int index = getIndex(key);
+        Node<K, V> lastNode = getLastCollisionNode(index);
+        if (lastNode == null) {
+            nodes[index] = newNode;
+            size++;
+        } else {
+            putNewNode(newNode, nodes[index]);
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int index = getIndex(key);
+        Node<K, V> firsNodeOnIndex = nodes[index];
+        if (firsNodeOnIndex == null) {
+            return null;
+        }
+        if (firsNodeOnIndex.next == null) {
+            return firsNodeOnIndex.value;
+        }
+        Node<K, V> foundNode = findNestedNode(firsNodeOnIndex, key);
+        return foundNode.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        if (key == null) {
+            return 0;
+        }
+        return Math.abs(key.hashCode() % capacity);
+    }
+
+    private Node<K, V> getLastCollisionNode(int index) {
+        if (nodes[index] == null) {
+            return null;
+        }
+        Node<K, V> lastNode = nodes[index];
+        while (lastNode.next != null) {
+            lastNode = lastNode.next;
+        }
+        return lastNode;
+    }
+
+    private Node<K, V> findNestedNode(Node<K, V> nextNode, K key) {
+        if (nextNode.next != null && !nextNode.equals(key)) {
+            nextNode = findNestedNode(nextNode.next, key);
+        }
+        return nextNode;
+    }
+
+    private void putNewNode(Node<K, V> newNode, Node<K, V> currentNode) {
+        K currentNodeKey = currentNode.key;
+        if (newNode.equals(currentNodeKey)) {
+            currentNode.value = newNode.value;
+            return;
+        }
+        if (currentNode.next == null) {
+            currentNode.next = newNode;
+            size++;
+            return;
+        }
+        putNewNode(newNode, currentNode.next);
+    }
+
+    private void resizeList() {
+        this.capacity = capacity * INCREASE_CAPACITY_VALUE;
+        this.threshold = (int) (DEFAULT_LOAD_FACTOR * capacity);
+        this.size = 0;
+        Node<K, V>[] prevNodes = nodes;
+        this.nodes = new Node[capacity];
+        for (Node<K, V> node : prevNodes) {
+            if (node != null) {
+                reassignNodesPositions(node);
+            }
+        }
+    }
+
+    private void reassignNodesPositions(Node<K, V> node) {
+        put(node.key, node.value);
+        if (node.next != null) {
+            reassignNodesPositions(node.next);
+        }
+    }
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+
+        @Override
+        public boolean equals(Object key) {
+            return Objects.equals(key, this.key);
+        }
```

Comment:
```suggestion
```

Reply:
what to here? both code look the same

Reply:
what to do here? codes look the same and deletion of the code causes errors